src/Bearded.TD/Game/Simulation/Elements/ArcTree.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bearded.TD.Game.Simulation.Damage;
+using Bearded.TD.Game.Simulation.GameObjects;
+using Bearded.TD.Game.Simulation.Navigation;
+using Bearded.TD.Game.Simulation.Weapons;
+using Bearded.TD.Tiles;
+using Bearded.TD.Utilities.Collections;
+using Bearded.Utilities.Linq;
+using Bearded.Utilities.SpaceTime;
+using TimeSpan = Bearded.Utilities.SpaceTime.TimeSpan;
+
+namespace Bearded.TD.Game.Simulation.Elements;
+
+interface IArcSource
+{
+    IEnumerable<IArcModifier> ContinuationModifiers { get; }
+
+    ArcTree.ArcContinuation GetInitialArcParametersWithoutModifiers();
+}
+
+interface IArcModifier
+{
+    void Modify(ref ArcTree.ArcContinuation arc);
+}
+
+static class ArcTree
+{
+    private record struct ArcHit(
+        GameObject Source,
+        GameObject Target,
+        ArcContinuation Continuation
+    );
+
+    public record struct ArcContinuation(
+        GameObject Source,
+        int BouncesBefore,
+        int BouncesLeft,
+        int Branches,
+        int MaxBounceDistance)
+    {
+        public static ArcContinuation Initial(
+            GameObject source, int bounces = 1, int branches = 1, int maxBounceDistance = 2)
+            => new(source, 0, bounces, branches, maxBounceDistance);
+    }
+
+    public static void Strike(
+        GameState game,
+        GameObject? source,
+        IArcSource arcSource,
+        ImmutableArray<Tile> tilesInRange,
+        UntypedDamage damage,
+        TimeSpan lifeTime)
+    {
+        var context = setupContext(game, source, arcSource);
+
+        hitInitialTargets(context, arcSource, tilesInRange);
```

Comment:
I tried refactoring it in several ways. Ultimately, I prefer keeping the entire state in a single object, which is now called State, implying (I hope) mutability. There are no more methods, so at least there's less magic. It behaves as if this was an object and you'd be calling `this.` instead of `state.` - but I prefer the static methods, rather than having all that code inside the same object as the state.

We could still decide to go this extra step and move the code into the state if we prefer later, and could then even expose it publicly and make it reusable to save on allocations. But I'll worry about that after profiling.